price-estimator/src/models/markets_results.rs

```diff
@@ -25,6 +26,34 @@ impl From<&pricegraph::TransitiveOrderbook> for MarketsResult {
     }
 }
 
+impl MarketsResult {
+    pub fn into_base_units(
+        self,
```

Comment:
Should this then also take a &mut self? What would be the naming convention for that?

Reply:
I think taking `self` here is fine, since it doesn't have any `Box`-ed data anyway. In that case you coud:
```
pub fn into_base_units(mut selt, ...) -> Self {
    transitive_orders_in_base_unit(&mut self.asks, ...);
    transitive_orders_in_base_unit(&mut self.bids, ...);
    self
}
```